glue/sample/src/sinter/_collection/_collection_manager.py

```diff
@@ -129,6 +129,10 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def start_workers(self, *, actually_start_worker_processes: bool = True):
         assert not self.started
 
+        # Use max_batch_size from collection_options if provided, otherwise default to 1024 as large
+        # batch sizes can lead to thrashing
```

Comment:
I pulled this comment from the README comment "setting batch sizes too large causes thrashing" but have not actually profiled myself

Reply:
For the Github record there is not much to be gained in most situations by moving away from the 4096 default

<img width="793" alt="image" src="https://github.com/user-attachments/assets/25af0526-ba57-414f-8e0e-07800b3e8c80" />

[snippet](https://gist.github.com/caryan/6b941df93b0598e32eef153bd997e8dd)